### PR TITLE
Handle invalid user/package names better

### PR DESCRIPTION
**Pull request by dimaryaz**

### CODE REVIEW

File: quilt/tools/command.py [hunk 1]
```diff
@@ -148,6 +148,12 @@ def _parse_package(name):
             raise ValueError
     except ValueError:
         raise CommandException("Specify package as owner/package_name.")
+
+    try:
+        PackageStore.check_name(owner, pkg)
+    except StoreException as ex:
+        raise CommandException(str(ex))
```
Comment (akarve) [inline quilt/tools/command.py]: so the upstream CommandException has a good error message that says "illegal characters" or something?

Comment (dimaryaz) [inline quilt/tools/command.py]: The error message stays the same ("Invalid package name: ...") - the only difference is, the `main` method catches `CommandException` and just prints out the message so you don't see the stack trace.

Comment (akarve) [inline quilt/tools/command.py]: the message should tell people how to do things right. so "Package name contains illegal characters. Rename package with alphanumerics and underscore..." or whatever. I'm not sure if we support numerics in trailing positions.

Comment (dimaryaz) [inline quilt/tools/command.py]: Ok, I'll do that in a separate PR. "Rename package" wouldn't be appropriate if e.g. you're running `quilt install ...`; in that case, you just got the name wrong, and there's nothing to rename.

**Pull request merged**